cdap-api-common/src/main/java/io/cdap/cdap/api/exception/ErrorCategory.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright © 2024 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.api.exception;
+
+/**
+ * Class representing the category of an error.
+ *
+ * <p>This class is used to classify errors into different categories,
+ * such as plugin errors, provisioning errors, etc.
+ * </p>
+ */
+public class ErrorCategory {
+  public static final String PLUGIN = "Plugin";
+  public static final String OTHERS = "Others";
+  private final boolean isPlugin;
+  private final String errorCategory;
+
+  /*
+   * Constructor for ErrorCategory.
+   *
+   * @param isPlugin {@code true} if the error is related to a plugin, {@code false} otherwise.
+   * @param errorCategory The category of the error.
+   */
+  public ErrorCategory(boolean isPlugin, String errorCategory) {
+    this.isPlugin = isPlugin;
+    this.errorCategory = errorCategory;
+  }
+
+  /*
+   * Constructor for ErrorCategory.
+   *
+   * @param isPlugin {@code true} if the error is related to a plugin, {@code false} otherwise.
+   */
+  public ErrorCategory(boolean isPlugin) {
+    this.isPlugin = isPlugin;
+    this.errorCategory = null;
+  }
+
+  /*
+   * Returns the category of the error.
+   */
+  public String getErrorCategory() {
+    return errorCategory == null ? (isPlugin ? PLUGIN : OTHERS) :
```

Comment:
Are we going with 2 categories only ? 
if it's more, we should have an Enum ?

Reply:
Introduced an error category `enum` and sub category as `string`.

Reply:
`ErrorCategoryEnum` can be extended as per requirements.